src/github_query_issues.py

```diff
@@ -8,128 +8,118 @@
 
 import requests
 import json
-import re
 import os
-from typing import Set, List
-from utils import ensure_folder_exists, save_state_to_json_file, initialize_request_session
-from containers import Repository, Issue
-
-OUTPUT_DIRECTORY = "../data/fetched_data/feature_data"
-ISSUE_PER_PAGE = 100
-
 
-def sanitize_filename(filename: str) -> str:
-    """
-    Sanitizes the provided filename by removing invalid characters and replacing spaces with underscores.
+from containers.issue import Issue
+from containers.repository import Repository
+from typing import Set, List
 
-    @param filename: The filename to be sanitized.
+from utils import ensure_folder_exists, save_issues_to_json_file, initialize_request_session
 
-    @return: The sanitized filename.
-    """
-    # Remove invalid characters for Windows filenames
-    sanitized_name = re.sub(r'[<>:"/|?*`]', '', filename)
-    # Reduce consecutive periods
-    sanitized_name = re.sub(r'\.{2,}', '.', sanitized_name)
-    # Reduce consecutive spaces to a single space
-    sanitized_name = re.sub(r' {2,}', ' ', sanitized_name)
-    # Replace space with '_'
-    sanitized_name = sanitized_name.replace(' ', '_')
 
-    return sanitized_name
+OUTPUT_DIRECTORY = "../data/fetched_data/issue_data"
+ISSUE_PER_PAGE = 100
 
 
-def save_issue_without_duplicates(issue: dict, loaded_issues: List[dict], added_issue_numbers: Set[int]) -> None:
+def filter_issues_by_numbers(issues: List[Issue], added_issue_numbers: Set[int]) -> (List[Issue], Set[int]):
```

Comment:
`issues` parameter is missing in docstring. 
Check TODO in a file.

Reply:
Documentation, comments and docstrings will be part of other issue. This PR is focused on bringing OOP into the project. This comment is irrelevant for this moment.